src/config/searchWorker.ts

```diff
@@ -16,29 +16,147 @@
+     * @returns A sorting function.
+     */
+    const dynamicSort = (property: string, sortOrder: SortingOrder) => {
+        return (a: Record<string, string | number>, b: Record<string, string | number>) => {
+            const valueA = a[property]
+            const valueB = b[property]
+
+            // Special cases handling where the property is not in sortable format.
+            if (Object.keys(propertyComparatorMap).includes(property)) {
+                return propertyComparatorMap[property](valueA, valueB, sortOrder)
+            }
+
+            // Handling of numbers and if one property is number and the other is string.
+            if (typeof valueA === 'number' || typeof valueB === 'number') {
+                return numberComparatorBySortOrder(
+                    typeof valueA === 'number' ? valueA : 0,
+                    typeof valueB === 'number' ? valueB : 0,
+                    sortOrder,
+                )
+            }
+
+            // Handling of strings and numbers in string type.
+            if (typeof valueA === 'string' && typeof valueB === 'string') {
+                if (!Number.isNaN(Number(valueA)) || !Number.isNaN(Number(valueB))) {
+                    return numberComparatorBySortOrder(
+                        !Number.isNaN(Number(valueA)) ? Number(valueA) : 0,
+                        !Number.isNaN(Number(valueB)) ? Number(valueB) : 0,
+                        sortOrder,
+                    )
+                }
+                return stringComparatorBySortOrder(valueA, valueB, sortOrder)
+            }
+
+            return 0
+        }
+    }
+
+    const debounceSearch = <T extends unknown[]>(callback: (...args: T) => void) => {
+        let timeout: NodeJS.Timeout
+        return (...args: T) => {
             clearTimeout(timeout)
             timeout = setTimeout(() => {
                 callback.apply(self, args)
             }, 300)
         }
     }
 
-    function getFilteredList({ searchText, list }) {
-        const searchTextLowerCased = searchText.toLowerCase()
-        if (searchTextLowerCased === '' || !list?.length) {
-            self.postMessage(list)
-            return
-        }
-        self.postMessage(
-            list.filter((item) =>
+    const getFilteredList = ({
+        searchText,
+        list,
+        sortBy,
+        sortOrder,
+    }: {
+        searchText: string
+        list: unknown[]
+        sortBy: string
+        sortOrder: SortingOrder
+    }) => {
+        const searchTextLowerCased = searchText.trim().toLowerCase()
+        let filteredList = [...list]
+
+        if (searchTextLowerCased !== '' && list?.length) {
+            filteredList = list.filter((item) =>
                 Object.entries(item).some(
                     ([key, value]) => key !== 'id' && String(value).toLowerCase().includes(searchTextLowerCased),
                 ),
-            ),
-        )
+            )
+        }
+
+        if (sortBy && sortOrder) {
```

Comment:
FYI, we are out of list.length check at line 147. check if it should be added for both in a single if

Reply:
discussed